api/views.py

```diff
@@ -815,3 +820,103 @@ def create(self, request, *args, **kwargs):
         except Exception as err:
             print(err)
         return HttpResponse("Created new template: {}.".format(name), status=200)
+
+
+@api_view(["GET", "POST"])
+@permission_classes(
+    (
+        IsAuthenticated,
+        AdminPermission,
+    )
+)
+def update_app_status(request):
+    """
+    Manages the app instance status.
+    Implemented as a DRF function based view.
+    Supports GET and POST verbs.
+
+    The service contract for the POST actions is as follows:
+    :param release str: The release id of the app instance.
+    :param new-status str: The new status code.
+    :param event-ts timestamp: A JSON-formatted timestamp, e.g. 2024-01-25T16:02:50.00Z.
+    :param event-msg json dict: An optional json dict containing pod-msg and/or container-msg.
+    :returns: An http status code and status text.
+    """
+
+    # POST verb
+    if request.method == "POST":
+        print("INFO: API method update_app_status called with POST verb.")
+
+        release = None
+        new_status = None
+        event_msg = None
+        event_ts = None
+
+        utc = pytz.UTC
+
+        try:
+            # Parse and validate the input
+
+            # Required input
+            release = request.data["release"]
+            new_status = request.data["new-status"]
+
+            event_ts = datetime.strptime(request.data["event-ts"], "%Y-%m-%dT%H:%M:%S.%fZ")
+            event_ts = utc.localize(event_ts)
+
+            # Optional
+            if "event-msg" in request.data:
+                event_msg = request.data["event-msg"]
+
+        except KeyError as err:
+            print(f"API method called with invalid input. Missing required input parameter: {err}")
+            return Response(f"Invalid input. Missing required input parameter: {err}", 400)
+
+        except Exception as err:
+            print(f"API method called with invalid input:  {err}, {type(err)}")
```

Comment:
I think that instead of printing exception it would be nice at some point to just use request id to see in logs what went wrong
https://github.com/dabapps/django-log-request-id

Reply:
I would love to use a real logging lib. As soon as we introduce logging then we can turn all of these print statements into logging calls. The DEBUG prefix is a hint of the logging level to use at that time.